awx/ui_next/src/screens/Team/TeamEdit/TeamEdit.jsx

```diff
@@ -1,72 +1,45 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { withRouter } from 'react-router-dom';
+import { withRouter, useHistory } from 'react-router-dom';
 import { CardBody } from '@components/Card';
 
 import { TeamsAPI } from '@api';
 import { Config } from '@contexts/Config';
 
 import TeamForm from '../shared/TeamForm';
 
-class TeamEdit extends Component {
-  constructor(props) {
-    super(props);
+function TeamEdit({ team }) {
+  const history = useHistory();
+  const [error, setError] = useState(null);
 
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleCancel = this.handleCancel.bind(this);
-    this.handleSuccess = this.handleSuccess.bind(this);
-
-    this.state = {
-      error: '',
-    };
-  }
-
-  async handleSubmit(values) {
-    const { team } = this.props;
+  const handleSubmit = async values => {
     try {
       await TeamsAPI.update(team.id, values);
-      this.handleSuccess();
+      history.push(`/teams/${team.id}/details`);
     } catch (err) {
-      this.setState({ error: err });
+      setError(error);
```

Comment:
The name of the thrown exception and the name of the value passed to `setError` should match.